### PR TITLE
Feature/readme file

### PR DESCRIPTION
**Pull request by SHGODE**
@mattdrayer Do you think the first 2-3 paragraphs are too much information to put on the readme file?

### CODE REVIEW

**Comment by mattdrayer** [comment]
I think the first 2-3 paragraphs could be written more for the open source community, and so they could reference the ECC project as the initial problem which needed to be solved, but most of the references should be to Open LXP and not to ECC.

**Review by KarenAJ — APPROVED** [review]
Approved!

**Comment by SHGODE** [comment]
> I think the first 2-3 paragraphs could be written more for the open source community, and so they could reference the ECC project as the initial problem which needed to be solved, but most of the references should be to Open LXP and not to ECC.

**Comment by SHGODE** [comment]
Hi, @mattdrayer  just added swagger.yaml file. Can you please confirm that the file path and file look good to you? Thank you!

File: swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+          description: Bad Request Sent
+          
+  /catalogs/:
+    summary: API to get catelogs list
+    description: ' Catelogs list'
+    get:
+      summary: Get A List Of Record Of Specified Catalogs
+      description: Get A List Of Record Of Specified Catalogs
+      tags:
+      - "Catelogs"
```
Comment (mattdrayer) [inline swagger.yaml:164]: "Catalogs"

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                  type: string
+      responses:
+        '201':
+          description: "New Record Created"        
+        '400':    # status code
+          description: Bad Request Sent
+          
+  /catalogs/:
+    summary: API to get catelogs list
+    description: ' Catelogs list'
```
Comment (mattdrayer) [inline swagger.yaml:159]: "Catalogs"

File: swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                type: array
+                items: 
+                  type: string
+        '500':    # status code
+          description: Internal Server Error       
+  
+    patch:
+      tags:
+      - "Metadata"
+      summary: "Add New Metadata Record"
```
Comment (mattdrayer) [inline swagger.yaml:104]: Typically PATCH is not used to add a new record, it's used to edit an existing record.

File: swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    summary: API to get records based on UUID
+    description: ' XIS api'
+    get:
+      summary: Get Record Based On Unique Record Identifier
+      description: Get Record Based On Unique Record Identifier
+      tags:
+      - "Metadata"
+      parameters:
+        - in: query
+          name: id
```
Comment (mattdrayer) [inline swagger.yaml:26]: is it `id` or `uuid`?

File: swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,200 @@
+openapi: 3.0.0
+info:
+  title: Swagger Experience Index Service APIs
+  version: 1.0.0
+  description: APIs For Access To Openlxp Experience Index Service Information
```
Comment (mattdrayer) [inline swagger.yaml:5]: "OpenLXP"

File: swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  contact:
+    email: "openlxphost@gmail.com"
```
Comment (mattdrayer) [inline swagger.yaml:7]: is this email valid?

File: swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+  license:
+    name: "Apache License"
```
Comment (mattdrayer) [inline swagger.yaml:9]: "Apache License, Version 2.0"

File: swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+    url: "http://www.apache.org/licenses/LICENSE-2."
```
Comment (mattdrayer) [inline swagger.yaml:10]: http://www.apache.org/licenses/LICENSE-2.0

File: swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+servers:
+  - url: http://localhost:8080/api/
+    variables: {}
+    description: XIS LocalHost
+paths:
+  /metadata/:
+    summary: API to get records based on UUID
+    description: ' XIS api'
```
Comment (mattdrayer) [inline swagger.yaml:18]: Is there an extra space at the beginning of this description?  Also "API" should be capitalized.

File: swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+      parameters:
+        - in: path
+          name: id
+          description: "Enter Unique ID To Fetch Metadata Record"
+          schema:
+            type: string
+          required: true
+      responses:
+        '200':    # status code
+          description: A JSON array of user names
```
Comment (mattdrayer) [inline swagger.yaml:91]: "user names" ?

File: swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+    summary: API to run ETL pipeline in XIS
+    description: ' XIS api'
+    get:
+      summary: Run XIS Workflow
+      description: Run XIS Workflow
+      tags:
+      - "XIS Workflow"
+      responses:
+        '200':    # status code
+          description: A JSON array of user names
```
Comment (mattdrayer) [inline swagger.yaml:189]: "user names" again here

File: swagger.yaml (hunk 4 above)
Comment (SHGODE) [reply swagger.yaml:26]: It's id in XSE, that is why we are using id here. But in general it's UUID.

File: swagger.yaml (hunk 6 above)
Comment (SHGODE) [reply swagger.yaml:7]: yes

**Review by mattdrayer — APPROVED** [review]
Ship it!

**Review by KarenAJ — APPROVED** [review]
Approved!

**Pull request merged**